client/question/question.js

```diff
@@ -550,101 +550,195 @@ tie.constant(
+          'declares <code>', varName, '</code> as an array variable. Another ',
+          'possibility is that <code>', varName, '</code> was misspelled or ',
+          'uses incorrect capitalization.'].join('');
+      }
+    }, {
+      friendlyErrorCheck: function(errorString) {
+        var re = new RegExp('^SyntaxError: invalid syntax');
+        return re.test(errorString);
+      },
+      getFriendlyErrorText: function() {
+        return [
+          'This is the default, generic syntax error message in Python. Here ',
+          'are some common mistakes that can result in this error:',
+          '<ul>',
+          '<li>Mis-matching (missing or too many) braces, brackets, ',
+          'parentheses, or quotation marks',
+          '<li>Missing colons when defining a function (e.g., ',
+          '<code>def my_function</code> should be ',
+          '<code>def my_function:</code>)',
+          '<li>Misspelled Python keywords (e.g., misspelling ',
+          '<code>for</code> in a for loop)',
+          '</ul>',
+          'Note that since this is the default, generic error, it could also ',
+          'be caused by something else.'].join('');
+      }
+    }, {
+      friendlyErrorCheck: function(errorString) {
+        var re = new RegExp('^SyntaxError: EOL while scanning string literal');
+        return re.test(errorString);
+      },
+      getFriendlyErrorText: function() {
+        return [
+          'An "End of Line" error on a string usually means you are ',
+          'missing a quotation mark somewhere.'].join('');
+      }
+    }, {
+      friendlyErrorCheck: function(errorString) {
+        var re = new RegExp('^IndentationError: ');
+        return re.test(errorString);
+      },
+      getFriendlyErrorText: function() {
+        return [
+          'A line of code might not be indented properly. Python is ',
+          'particular about indentation, since that\'s how it identifies ',
+          'code blocks.'].join('');
+      }
+    }, {
+      friendlyErrorCheck: function(errorString) {
+        var re = new RegExp('^TypeError: can only concatenate');
+        return re.test(errorString);
+      },
+      getFriendlyErrorText: function() {
+        return [
+          'You may have tried to combine two incompatible types (you might ',
+          'have tried to combine a string with an array, etc.).'].join('');
+      }
+    }
+  ]
+});
+
+/**
+ * Dictionary of the novice-friendly runtime error feedback messages
+ *
+ * `friendlyErrorCheck` function checks to see if the error matches with the
+ * type of feedback message
+ *
+ * `getFriendlyErrorText` function returns the appropriate feedback message for
+ * the given error
+ *
+ * @type {Object}
  **/
-tie.constant('RUNTIME_ERROR_FEEDBACK_MESSAGES', {
+tie.constant('FRIENDLY_RUNTIME_ERROR_TRANSLATIONS', {
   python: [{
     // Error when user indents lines incorrectly
-    checker: function(errorString) {
+    friendlyErrorCheck: function(errorString) {
```

Comment:
This is interesting, it appears in the syntax error checks too. Should we remove it in one of them? If not, should the error messages be consistent?

(If de-duplicating, it might be worth noting that the error messages used in this section have already undergone language review.)

Reply:
I noticed that too, but I viewed it as out of scope for this (rather lengthy)PR (i.e., revisit how runtime errors work versus syntax and update translations appropriately). For now, I can at least make them consistent.

Reply:
SG re consistency.

Reply:
Done in next commit.